src/openms/include/OpenMS/ANALYSIS/OPENSWATH/MRMFeatureFilter.h

```diff
@@ -137,55 +134,80 @@ namespace OpenMS
       @param meta_value_u Upper bound (inclusive) for the metaValue range
 
       @return True if the metaValue is within the bounds, and False otherwise.
-    */ 
-    bool checkMetaValue(const Feature & component, const String & meta_value_key, const double & meta_value_l, const double & meta_value_u);
-    
+    */
+    bool checkMetaValue(
+      const Feature& component,
+      const String& meta_value_key,
+      const double& meta_value_l,
+      const double& meta_value_u
+    ) const;
+
     /**
       @brief Count the number of heavy/light labels and quantifying/detecting/identifying transitions
 
       @param component component_group with subordinates
       @param transitions transitions from a TargetedExperiment
 
       @return Map of labels/transition types and their corresponding number.
-    */ 
-    std::map<String,int> countLabelsAndTransitionTypes(const Feature & component_group,
-      const TargetedExperiment & transitions);
-    
+    */
+    std::map<String, Int> countLabelsAndTransitionTypes(
+      const Feature& component_group,
+      const TargetedExperiment& transitions
+    ) const;
+
     /**
       @brief Sorts, removes duplicates, and concatenates a list of Strings
 
       @param str_vec vector of Strings
       @param delim token to seperate Strings in the list
 
       @return A concatenated string.
-    */ 
-    String uniqueJoin(std::vector<String>& str_vec, String& delim);
+    */
+    String uniqueJoin(std::vector<String>& str_vec, const String& delim) const;
 
-private:
-    template <typename T>
-    bool checkRange(T const& value, T const& value_l, T const& value_u);
+    /**
+      @brief Tests whether the given value is in the range [value_l, value_u].
+
+      @tparam T Type of the given inputs (e.g., `double`)
 
-    // Members
+      @param[in] value Tested value
+      @param[in] value_l Lower bound
+      @param[in] value_u Upper bound
 
-    /// flag or filter (i.e., remove) features that do not pass the QC
+      @return `true` if the value is in the range `[value_l, value_u]`,
+      `false` otherwise.
+    */
+    template <typename T>
+    bool isWithinRange(const T& value, const T& value_l, const T& value_u) const;
+
+protected:
+    /// The number of QC metrics for a single component
+    Size N_C_QC_METRICS = 4;
```

Comment:
Are you updating these values somehow?  The metavalue will be a dynamic number of metrics, so I am not sure this is a good idea.